### PR TITLE
updated Guide and fixed tex

### PR DESCRIPTION
**Pull request by alinchis**

### CODE REVIEW

File: assets/icons.html [hunk 1]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
+=======
```
Comment (shashi) [inline assets/icons.html]: this stuff is still around

File: README.md [hunk 2]
```diff
@@ -203,7 +199,7 @@ end
 
 ---------------------------------------------
 
-![Empty Tile Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/empty_tile.png "Empty Tile")
+![Empty Tile Output](./assets/img/empty_tile.png "Empty Tile")
```
Comment (shashi) [inline README.md]: use https://cloud.githubusercontent.com/assets/25916/15371154/d15b3bc2-1d57-11e6-8d41-1f2e5d535d78.png this URL, and you can remove the file from the repo

File: README.md [hunk 3]
```diff
@@ -261,55 +257,84 @@ end
...
+
+    tex(L"f(x) = \int_{-\infty}^\infty \hat f(\xi)\,e^{2 \pi i \xi x} \,d\xi")
+end
+```
+
+**Output:**
+
+---------------------------------------------
+
+![TeX Tile Output](./assets/img/tex_tile.png "TeX Tile")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371196/03cd08ec-1d58-11e6-90af-b282502ad565.png

File: README.md [hunk 4]
```diff
@@ -261,55 +257,84 @@ end
...
+    tex(latexstring(a, " ",b))
 end
 ```
 
 **Output:**
 
 ---------------------------------------------
 
-![TeX Tile Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/tex_tile.png "TeX Tile")
+![TeX Tile Output](./assets/img/tex_tile.png "TeX Tile")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371196/03cd08ec-1d58-11e6-90af-b282502ad565.png

File: README.md [hunk 5]
```diff
@@ -261,55 +257,84 @@ end
...
+        x=linspace(-8,8,150), y=linspace(-2,2,150), Geom.contour)
 end
 ```
 
 **Output:**
 
 ---------------------------------------------
 
-![Gadfly Tile Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/gadfly_tile.png "Gadfly Tile")
+![Gadfly Tile Output](./assets/img/gadfly_tile.png "Gadfly Tile")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371247/378a2890-1d58-11e6-8b1b-d5f44676d11e.png

File: README.md [hunk 6]
```diff
@@ -337,28 +362,28 @@ end
 
 ---------------------------------------------
 
-![Compose Tile Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/compose_tile.png "Compose Tile")
+![Compose Tile Output](./assets/img/compose_tile.png "Compose Tile")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371280/4f11b08c-1d58-11e6-882b-73ab0fc69d9b.png

File: README.md [hunk 7]
```diff
@@ -337,28 +362,28 @@ end
...
+    fillcolor("pink", Escher.pad(1em, "Simple text"))
 end
 ```
 
 **Output:**
 
 ---------------------------------------------
 
-![Text Tile Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/text_tile.png "Text Tile")
+![Text Tile Output](./assets/img/text_tile.png "Text Tile")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371304/6a17736c-1d58-11e6-9882-704d2d9813d1.png

File: README.md [hunk 8]
```diff
@@ -388,30 +413,37 @@ function main(window)
...
 ```
-`x` and `y` are vertial arrangements of 4 tiles each, these arrangements are themselves put in a `hbox` to place `x` next to `y`.
+`x` and `y` are vertical arrangements of 4 tiles each, these arrangements are themselves put in a `hbox` to place `x` next to `y`.
 
 **Output:**
 
 ---------------------------------------------
 
-![Basic Layout Tile Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/basic_layout_tile.png "Basic Layout Tile")
+![Basic Layout Tile Output](./assets/img/basic_layout_tile.png "Basic Layout Tile")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371323/818a7404-1d58-11e6-8c6b-0bec77fdc369.png

File: README.md [hunk 9]
```diff
@@ -430,18 +462,17 @@ function main(window)
...
-For more icon graphics visit the [Polymer Catalog](https://elements.polymer-project.org/elements/iron-icons?view=demo:demo/index.html&active=iron-icons). Choose from the Icons section the one you like, the format is `icons:icon_name` and use it by calling `icon("icon_name")`.
-
 The `pages` function combines tiles into a set of pages - only a single page is visible at a time. Pages can be combined with `tabs` to allow switching between pages.
 
 **Output:**
 
 ---------------------------------------------
 
-![Higher Order Layout Tile Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/higher_order_layout_tile.png "Higher Order Layout Tile")
+![HO Layout Tile Output](./assets/img/ho_layout_tile.gif "Higher Order Layout Tile")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371341/94f1545e-1d58-11e6-97ca-d94bd1e90afd.png

File: README.md [hunk 10]
```diff
@@ -520,17 +551,20 @@ The `connected_slider` renders as a slider and updates the signal iterations whe
 
 ---------------------------------------------
 
-![Interactive Tile 1 Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/interactive_tile_1.png "Interactive Tile 1")
+![Interactive Tile 1 Output](./assets/img/interactive_tile.png "Interactive Tile 1")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371366/ab7026f6-1d58-11e6-9037-e9107f810e06.png

File: README.md [hunk 11]
```diff
@@ -557,12 +591,262 @@ end
 
 ---------------------------------------------
 
-![Interactive Tile 2 Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/interactive_tile_2.png "Interactive Tile 2")
+![Interactive Tile 2 Output](./assets/img/interactive_ui.png "Interactive Tile 2")
```
Comment (shashi) [inline README.md]: https://cloud.githubusercontent.com/assets/25916/15371377/c032aeec-1d58-11e6-8af9-2017d77a94c3.png

File: src/library/tex.jl [hunk 12]
```diff
@@ -1,14 +1,16 @@
+using LaTeXStrings
```
Comment (shashi) [inline src/library/tex.jl]: Add this to REQUIRE?

File: README.md [hunk 13]
```diff
@@ -134,12 +130,12 @@ For your reference, the default HTML content of the `assets/` folder is listed b
 | no | HTML file | content | API | loading-time |
```
Comment (shashi) [inline README.md]: I really like this table, great work!

File: README.md [hunk 14]
```diff
@@ -557,12 +591,262 @@ end
...
 Escher provides ready-to-use `Tiles` like buttons, input, codemirror etc., in the [Widgets API](http://escher-jl.org/widgets-api.html). Keep in mind that, regular and interactive, `Tiles` can be augmented with clickable, selectable, keypress etc. behavior via the [Behavior API](http://escher-jl.org/behavior-api.html).
 
+
 ### 4.3. Escher Advanced
+
+This section is intended to provide a more detailed view on the way Escher works and how to create your own custom assets.
+
+#### 4.3.1. Assets basics
+
+I can think of three reasons why one might need to create new assets, for fun not included:
```
Comment (shashi) [inline README.md]: It's kind of unclear what an asset means to someone who's not familiar with the term in the context of web development.

File: README.md [hunk 15]
```diff
@@ -557,12 +591,262 @@ end
...
 ---------------------------------------------
 
 ##### 4.2.5.3. Interactive Tiles
 
 Escher provides ready-to-use `Tiles` like buttons, input, codemirror etc., in the [Widgets API](http://escher-jl.org/widgets-api.html). Keep in mind that, regular and interactive, `Tiles` can be augmented with clickable, selectable, keypress etc. behavior via the [Behavior API](http://escher-jl.org/behavior-api.html).
 
+
 ### 4.3. Escher Advanced
+
+This section is intended to provide a more detailed view on the way Escher works and how to create your own custom assets.
```
Comment (shashi) [inline README.md]: Can just say "custom tile" here

File: README.md [hunk 16]
```diff
@@ -557,12 +591,262 @@ end
...
+
+#### 4.3.1. Assets basics
+
+I can think of three reasons why one might need to create new assets, for fun not included:
+
+- if the current APIs do not provide the elements you require, you can build your own.
+- Escher may not give access to all available properties defined in the HTML element, or it may have certain default values that you would like to change.
+- you might want to optimize your code. When loading assets, all templates in the assets package get loaded into the webpage, even the ones that are not used. For the development phase this is not an issue, but in production you might want to provide a faster loading time by creating your package only with the templates you are using.
+
+To create a new asset element you need three things:
```
Comment (shashi) [inline README.md]: asset element should really be a "custom tile"

File: README.md [hunk 17]
```diff
@@ -557,12 +591,262 @@ end
...
+
+1. a custom `HTML` element template you wish to use
```
Comment (shashi) [inline README.md]: "custom HTML element template" -> "custom HTML element" (link to http://www.html5rocks.com/en/tutorials/webcomponents/customelements/)

File: README.md [hunk 18]
```diff
@@ -557,12 +591,262 @@ end
...
+2. a `Tile` type to match the HTML element kind, depending on what kind of interaction is needed. The `Tile` fields do not necessarily mirror the properties of the HTML.
```
Comment (shashi) [inline README.md]: "A `Tile` which corresponds to the custom element" is more clear. can leave out the phrase ", depending on what kind of interaction is needed. "

**Comment by alinchis** [comment]
I was thinking at the asset file, which can contain multiple `Tiles`

File: README.md (hunk 2 above)
Comment (alinchis) [inline README.md]: ok

File: README.md [hunk 19]
```diff
@@ -557,12 +591,262 @@ end
...
+3. a `render(Tile, state) = Elem(HTML element)` function that does the wiring. The user chooses which `Tile` fields correspond to which `HTML` property.
```
Comment (shashi) [inline README.md]: this is till complex to understand. You can just say,  "a `render(::MyTile, state)` which returns an `Elem` (link to https://github.com/shashi/Patchwork.jl#creating-elements). "This may create the custom element you created in step 1"?

File: README.md [hunk 20]
```diff
@@ -557,12 +591,262 @@ end
...
+
+These three requirements are placed in two different files:
+
+1. `assets/<asset-name>.html` is where you place the HTML code
+2. `src/library/<asset-name>.jl` holds the custom `Tile` and the `render` definition
+
+For usage, the `<asset-name>.html` is pushed to the browser from the `main` function inside your Julia Escher UI file: `push!(window.assets, "asset-name")`.
+
+On the other hand, `<asset-name>.jl` must be imported in the `src/Escher.jl` file: `include("library/<asset-name>.jl")`. A restart of Escher server is necessary in order to recompile with the new addition.
```
Comment (shashi) [inline README.md]: excellent instructions

File: README.md [hunk 21]
```diff
@@ -557,12 +591,262 @@ end
...
+    typedarg(side::Array{Sides}=[left,right,top,bottom])
+    curry(tile::Tile)
+    kwarg(color::Color=colorant"black")
+end
+```
+
+
+**Generated definitions**
+
+---------------------------------------------
```
Comment (shashi) [inline README.md]: are you using these lines consistently? it's fine though

File: assets/bower_components/iron-icon/iron-icon.html [hunk 22]
```diff
@@ -8,6 +8,7 @@
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 -->
 
+
```
Comment (shashi) [inline assets/bower_components/iron-icon/iron-icon.html]: would be nice to not touch these files

**Comment by shashi** [comment]
Could you squash these commits.

I'd say 2 commits are enough: 1 with the user guide changes another with the TeX fixes. (follow my commands on gitter ;))

File: src/library/tex.jl (hunk 12 above)
Comment (rohitvarkey) [inline src/library/tex.jl]: Yes. Travis is failing cause this isn't present in REQUIRE.

File: README.md (hunk 18 above)
Comment (shashi) [inline README.md]: well you can mention that an asset file can contain multiple custom element definitions somewhere. The organization of custom elements in .html files is orthogonal to the elements or the corresponding Tile types...

File: README.md (hunk 21 above)
Comment (alinchis) [inline README.md]: I used the lines to point out the output of the examples, and in this case the result of the @api macro

File: assets/bower_components/iron-icon/iron-icon.html (hunk 22 above)
Comment (alinchis) [inline assets/bower_components/iron-icon/iron-icon.html]: I messed with them to find out why they don't work. Unfortunately, I did not think in advance to try this in a different branch. There are no intentional updates to them.

File: src/library/tex.jl (hunk 12 above)
Comment (alinchis) [inline src/library/tex.jl]: I will add it now

File: assets/bower_components/iron-icon/iron-icon.html (hunk 22 above)
Comment (shashi) [inline assets/bower_components/iron-icon/iron-icon.html]: you can just leave them out when you do `git add -p`
